tools/ncbi_blast_plus/ncbi_macros.xml

```diff
@@ -258,35 +417,71 @@
     </xml>
+
     <xml name="input_filter_query_default_true">
         <param name="filter_query" argument="-seg" type="boolean" label="Filter out low complexity regions (with SEG)" truevalue="-seg yes" falsevalue="-seg no" checked="true" />
     </xml>
+
     <xml name="input_max_hits">
         <param name="max_hits" type="integer" min="0" value="0" label="Maximum hits to show" help="Use zero for default limits" />
         <param argument="-max_hsps" type="integer" min="1" optional="true" value="" label="Maximum number of HSPs (alignments) to keep for any single query-subject pair" help="The HSPs shown will be the best as judged by expect value. If this option is not set, BLAST shows all HSPs meeting the expect value criteria" />
     </xml>
+
     <xml name="input_evalue">
         <param name="evalue_cutoff" argument="-evalue" type="float" size="15" value="0.001" label="Set expectation value cutoff" />
     </xml>
+
     <xml name="input_word_size">
-        <param argument="-word_size" type="integer" min="0" value="0" label="Word size for wordfinder algorithm" help="Use zero for default, otherwise minimum 2" />
+        <param argument="-word_size" type="integer" min="2" optional="true"
+        label="Word size for wordfinder algorithm"
+        help="Leave blank for default, otherwise minimum 2" />
     </xml>
+
     <xml name="input_strand">
         <param argument="-strand" type="select" label="Query strand(s) to search against database/subject">
             <option value="-strand both">Both</option>
             <option value="-strand plus">Plus (forward)</option>
             <option value="-strand minus">Minus (reverse complement)</option>
         </param>
     </xml>
+
     <xml name="input_qcov_hsp_perc">
         <param argument="-qcov_hsp_perc" type="float" value="0" min="0" max="100" label="Minimum query coverage per hsp (percentage, 0 to 100)" help="See also the output column qcovhsp"/>
     </xml>
+
+    <xml name="input_window_size">
+        <param argument="-window_size" type="integer" optional="true" min="0"
+            label="Multiple hits window size: use 0 to specify 1-hit algorithm, leave blank for default"
+            help="Default window size varies with substitution matrix and BLAST type. Enter an integer to override the default."/>
+    </xml>
+
+    <xml name="input_threshold">
+        <param argument="-threshold" type="integer" optional="true" min="0"
+            label="Minimum score to add a word to the BLAST lookup table."
+            help="Leave blank for default, which varies based on application."/>
+    </xml>
+
+    <xml name="input_comp_based_stats">
+        <param argument="-comp_based_stats" type="select"
+            label="Composition-based statistics"
+            help="The default value varies based on application and task. Most common default is 2.">
+            <option value="-1">Use default</option>
```

Comment:
How about using a non-integer value for the default, e.g. the empty string or ``"default"`` which would be explicit?

Reply:
Using the blank string you could have the command token macro be consistent with most of the others.